### PR TITLE
Test chost errata pagination

### PR DESCRIPTION
**Pull request by swadeley**
Hello

Closed loop test for:
[BZ#1662254 – Per page value does not work for Content Host Errata page on Satellite WebUI](https://bugzilla.redhat.com/show_bug.cgi?id=1662254)

[Bug 1846670 - Unable to apply all Errata via Remote Execution on Web UI with "Select All" ](https://bugzilla.redhat.com/show_bug.cgi?id=1846670)

This test depends on these PRs:
[ Replace yum3 repo #7 ](https://github.com/SatelliteQE/fedorapeople-repos/pull/7)
[ Add pagination support on CHost's errata tab #532 ](https://github.com/SatelliteQE/airgun/pull/532)
[ Add support to install all CHost's errata #535 ](https://github.com/SatelliteQE/airgun/pull/535)

Test to follow soon.

### CODE REVIEW

**Comment by swadeley** [comment]
robottelo]$ pytest  --show-capture=no tests/foreman/ui/test_errata.py -k test_content_host_errata_page_pagination
============================== test session starts ======================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo_2/robottelo, inifile: pytest.ini
plugins: services-1.3.1, ibutsu-1.1.1, mock-1.10.4, forked-1.3.0, cov-2.10.1, xdist-1.34.0
collecting ... 2020-11-13 12:35:32 - conftest - DEBUG - Collected 15 test cases
collected 15 items / 14 deselected / 1 selected                                                                  

tests/foreman/ui/test_errata.py .                                                                          [100%]

=============== 1 passed, 14 deselected, 6 warnings in 494.45s (0:08:14) ====================

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Comment by swadeley** [comment]
Hello

Unfortunately, after updates in https://github.com/SatelliteQE/airgun/pull/532, this test now fails here:

                    # Go to CHost's Errata page and read the page's pagination widgets
     >               page_values = session.contenthost.read(client.hostname, widget_names=['errata.pagination'])

    tests/foreman/ui/test_errata.py:305:

**Comment by mshriver** [comment]
@swadeley 'Draft' state for PRs is better than using 'DO NOT MERGE' label, as it prevents the PR from being merged and communicates to reviewers that the PR isn't ready for review yet.

**Comment by swadeley** [comment]
> @swadeley 'Draft' state for PRs is better than using 'DO NOT MERGE' label, as it prevents the PR from being merged and communicates to reviewers that the PR isn't ready for review yet.

From my point of view it is ready for review because it was, then I was asked to make a change, and that change broke it. But I can address the reviews directly.

**Comment by mirekdlugosz** [comment]
I spent some time investigating this new failure today and I was able to pinpoint problem to updated widgetastic.core package. `Pagination.per_page` worked just fine on widgetastic 0.51, but fails on 0.60. See https://github.com/SatelliteQE/airgun/issues/536 for more details.

My suggestion is: continue development of this test case on Python venv that has widgetastic.core==0.51. Do note such venv must use Python 3.8 or earlier, as Python 3.9 removed `HTMLParser().unescape`, and old widgetastic depends on it.

Pagination issue needs be addressed separately, in Airgun or widgetastic. I wouldn't expect @swadeley to fix the problem just because he happened to be the first person to encounter this issue. I also wouldn't block his PR, for the same reason.

**Comment by swadeley** [comment]
@mirekdlugosz thank you, I confirm the test passes with the after 'pip install widgetastic.core==0.51'.

I have updated and pushed my change to make use of errata_id == "All" from https://github.com/SatelliteQE/airgun/pull/535#pullrequestreview-531291623.

Tests running again after the force push

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/ui/test_errata.py -k test_content_host_errata_page_pagination
================================ test session starts ===========================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo_2/robottelo, inifile: pytest.ini
plugins: ibutsu-1.12, mock-3.3.1, forked-1.3.0, services-2.2.1, cov-2.10.1, xdist-1.34.0
collecting ... 2020-11-20 14:34:37 - conftest - DEBUG - Collected 15 test cases
collected 15 items / 14 deselected / 1 selected                                                                  

tests/foreman/ui/test_errata.py .                                                                          [100%]

================ 1 passed, 14 deselected, 6 warnings in 557.73s (0:09:17) =============================

File: tests/foreman/ui/test_errata.py
Comment (mirekdlugosz) [inline tests/foreman/ui/test_errata.py:313]: Usually, we try to avoid using Airgun internals (like navigation and specific widgets) inside Robottelo. Sometimes there's no other way, but here it looks like `update()` and `read()` should suffice?

```python
page_values = session.contenthost.read(client.hostname, widget_names=['errata.pagination'])
assert int(page_values['errata']['pagination']['per_page']) == 20
session.contenthost.update({'errata.pagination.per_page': '100'})
```

Comment (mirekdlugosz) [inline tests/foreman/ui/test_errata.py:313]: Is that enough to cover 1662254? Shouldn't we first set `per_page` value, and then call `read()` on table to confirm that we see all errata? Or maybe call `read()` on pagination widget and assert there's only one page now.

File: tests/foreman/ui/test_errata.py
Comment (mirekdlugosz) [inline tests/foreman/ui/test_errata.py:281]: bz#184667 is about AMD processor support in RHEL5. Is that the correct number?

**Review by mirekdlugosz — COMMENTED** [review]
Now that we understand what major problem is about, I focused on some details in test itself.

**Comment by mirekdlugosz** [comment]
@swadeley also, please take a look at Github Actions / Travis results, as black is failing there.

File: tests/foreman/ui/test_errata.py
Comment (swadeley) [reply tests/foreman/ui/test_errata.py:281]: 1846670, missing 0, copy paste fail

File: tests/foreman/ui/test_errata.py
Comment (swadeley) [reply tests/foreman/ui/test_errata.py:313]: This does not work:
session.contenthost.update({'errata.pagination.per_page': '100'})
TypeError: update() missing 1 required positional argument: 'values'

Comment (mshriver) [reply tests/foreman/ui/test_errata.py:313]: @swadeley You should be able to resolve that by passing the dict as `values` kwarg.

Comment (swadeley) [reply tests/foreman/ui/test_errata.py:313]: Hi, yes figured out it probably should be :
 `session.contenthost.update(client.hostname, {'errata.pagination.per_page': '100'})`
but then that erros:
AttributeError: Object <<class 'airgun.widgets.Pagination'>> has no widget name "per_page"

**Review by mshriver — APPROVED** [review]
Changes look good to me, would like to see some test results paired with the repos container change in https://github.com/SatelliteQE/fedorapeople-repos/pull/7/files before I merge and deploy that.

File: tests/foreman/ui/test_errata.py
Comment (swadeley) [reply tests/foreman/ui/test_errata.py]: @mirekdlugosz I made some updates to that section

Comment (mirekdlugosz) [reply tests/foreman/ui/test_errata.py]: @swadeley that's probably because widgets returned by decorators are not properly included within `Widget.widget_names`. I wouldn't lose sleep over it, if `update()` is not good enough in this case, then it's not good enough in this case.

**Review by mirekdlugosz — APPROVED** [review]
@swadeley please fix Travis/GitHub Actions failures and I think we are good to merge.
I think your test results already include https://github.com/SatelliteQE/fedorapeople-repos/pull/7 , as without this change this test fails during setup?

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/ui/test_errata.py -k test_content_host_errata_page_pagination
============================= test session starts ===============================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo_2/robottelo, inifile: pytest.ini
plugins: ibutsu-1.12, mock-3.3.1, forked-1.3.0, services-2.2.1, cov-2.10.1, xdist-1.34.0
collecting ... 2020-11-24 16:04:17 - conftest - DEBUG - Collected 15 test cases
collected 15 items / 14 deselected / 1 selected                                                                  

tests/foreman/ui/test_errata.py .                                                                          [100%]


============================ 1 passed, 14 deselected, 7 warnings in 472.07s (0:07:52) ======================

File: tests/foreman/ui/test_errata.py
Comment (swadeley) [reply tests/foreman/ui/test_errata.py]: > @swadeley please fix Travis/GitHub Actions failures and I think we are good to merge.

@mirekdlugosz Change made (it was the dreaded `Reorder python imports` failure).

> I think your test results already include [SatelliteQE/fedorapeople-repos#7](https://github.com/SatelliteQE/fedorapeople-repos/pull/7) , as without this change this test fails during setup?

Correct, all test runs have always included the use of the new yum3 repo

Thank you

**Comment by swadeley** [comment]
> @swadeley also, please take a look at Github Actions / Travis results, as black is failing there.

Correct, all test runs have always included the use of the new yum3 repo

Thank you

File: tests/foreman/ui/test_errata.py
Comment (mshriver) [inline tests/foreman/ui/test_errata.py:259]: This test is missing required testimony tokens, which is what's failing in CI:

```
2020-11-24T16:11:27.3394966Z testimony --config testimony.yaml validate tests/foreman/ui
2020-11-24T16:11:33.2431256Z tests/foreman/ui/test_errata.py
2020-11-24T16:11:33.2432205Z ===============================
2020-11-24T16:11:33.2432675Z 
2020-11-24T16:11:33.2433578Z test_content_host_errata_page_pagination:259
2020-11-24T16:11:33.2435476Z --------------------------------------------
2020-11-24T16:11:33.2436095Z 
2020-11-24T16:11:33.2437811Z * Docstring should have at least caseautomation, casecomponent, caseimportance, caselevel, id, requirement, testtype, upstream token(s)
2020-11-24T16:11:33.2439386Z 
```

**Review by mshriver — CHANGES_REQUESTED** [review]
Failing CI due to missing testimony tokens, resolving that I'm happy to merge this along with the fedorapeople-repos PR.  We'll discuss deploying that container update when they're merged.

Meanwhile I've updated wt.core requirement in #8198

File: tests/foreman/ui/test_errata.py
Comment (swadeley) [reply tests/foreman/ui/test_errata.py:259]: @mshriver Fixed now. I think my git hooks must have been broken in this branch as even the UUID was missing,  and that is normally checked by pre-commit hooks.

**Review by mshriver — APPROVED** [review]

**Comment by swadeley** [comment]
Thank you

**Pull request merged**
